src/routes/view-content/bible-menu/BookChapterSelectorPane.svelte

```diff
@@ -77,66 +79,95 @@
+                firstSelectedVerse.number < verse.number)
+        ) {
+            lastSelectedVerse = { ...verse, selected: true };
+            currentBook.chapters = currentBook.chapters.map((chapter) => {
                 if (chapter.number === verse.chapterNumber) {
                     chapter.verseState = chapter.verseState.map((v) => {
-                        if (v.number === verse.number && v.chapterNumber === verse.chapterNumber) {
-                            return { ...v, selected: !v.selected };
+                        if (v.number === verse.number) {
+                            return { ...v, selected: true };
                         }
                         return v;
                     });
                 }
                 return chapter;
             });
 
-            currentBook = { ...currentBook, chapters: newChapters };
-            // needed to rerender the view
-            currentChapter = currentBook.chapters.find((c) => c.number === currentChapter?.number)!;
+            currentChapter = currentBook.chapters.find((c) => c.number === verse.chapterNumber) || null;
             return;
         }
 
-        if (selectedVerses.length > 0) {
-            const firstSelectedVerse = selectedVerses[0]!;
-
-            if (selectedVerses.length === 1) {
-                currentBook.chapters = currentBook.chapters.map((chapter) => {
-                    if (chapter.number === verse.chapterNumber) {
-                        chapter.verseState = chapter.verseState.map((v) => {
-                            if (v.number >= firstSelectedVerse.number && v.number <= verse.number) {
-                                return { ...v, selected: true };
-                            }
-                            return v;
-                        });
-                    }
-                    return chapter;
+        if (firstSelectedVerse && lastSelectedVerse) {
+            firstSelectedVerse = null;
+            lastSelectedVerse = null;
+            currentBook.chapters = currentBook.chapters.map((chapter) => {
+                chapter.verseState = chapter.verseState.map((v) => {
+                    return { ...v, selected: false };
                 });
-            } else {
-                const lastSelectedVerse = selectedVerses[selectedVerses.length - 1]!;
-                currentBook.chapters = currentBook.chapters.map((chapter) => {
-                    if (chapter.number === verse.chapterNumber) {
-                        chapter.verseState = chapter.verseState.map((v) => {
-                            if (v.number >= firstSelectedVerse.number && v.number <= lastSelectedVerse.number) {
-                                return { ...v, selected: false };
-                            }
-                            return v;
-                        });
-                    }
-                    return chapter;
-                });
-            }
+                return chapter;
+            });
```

Comment:
Do we need all of the `selected` attribute now or can that be ripped out since the `isVerseInSelectedRange` should handle the logic for that?